### PR TITLE
fix(EMS-2556-2570): No PDF - Multiple contract policy - Total months of cover - decimal places

### PR DESCRIPTION
**Pull request by ttbarnes**
## Introduction :pencil2:
This PR fixes an issue where the "Total months of cover" field of a no PDF application, would allow decimal points and cause a "problem with service" issue.

## Resolution :heavy_check_mark:
- Add a validation rule so that "Total months of cover" cannot be submitted with decimal points.
- Add E2E test coverage.

## Miscellaneous :heavy_plus_sign:
- Bump depedencies.
- Minor destructuring improvements.

### CODE REVIEW

File: e2e-tests/insurance/cypress/e2e/journeys/policy/multiple-contract-policy/validation/multiple-contract-policy-validation-total-months-of-cover.spec.js [hunk 1]
```diff
@@ -89,6 +89,18 @@ context('Insurance - Policy - Multiple contract policy page - form validation -
     });
   });
 
+  describe('when total months of cover contains a decimal', () => {
+    it('should render a validation error', () => {
+      cy.submitAndAssertFieldErrors(
+        field,
+        '7.5',
```
Comment (Zainzzkk) [inline e2e-tests/insurance/cypress/e2e/journeys/policy/multiple-contract-policy/validation/multiple-contract-policy-validation-total-months-of-cover.spec.js:96]: constant?

File: src/ui/server/controllers/insurance/policy/multiple-contract-policy/validation/rules/total-months-of-cover.test.ts [hunk 2]
```diff
@@ -56,6 +54,18 @@ describe('controllers/insurance/policy/multiple-contract-policy/validation/rules
     });
   });
 
+  describe('when a value contains a decimal', () => {
+    it('should return a validation error', () => {
+      mockBody[FIELD_ID] = '7.5';
```
Comment (Zainzzkk) [inline src/ui/server/controllers/insurance/policy/multiple-contract-policy/validation/rules/total-months-of-cover.test.ts:59]: does 7.5 have to be a test mock?

File: e2e-tests/insurance/cypress/e2e/journeys/policy/multiple-contract-policy/validation/multiple-contract-policy-validation-total-months-of-cover.spec.js (hunk 1 above)
Comment (ttbarnes) [reply e2e-tests/insurance/cypress/e2e/journeys/policy/multiple-contract-policy/validation/multiple-contract-policy-validation-total-months-of-cover.spec.js:96]: What kind of a constant? I don't think we shouldn't constants for plain numbers?

Comment (Zainzzkk) [reply e2e-tests/insurance/cypress/e2e/journeys/policy/multiple-contract-policy/validation/multiple-contract-policy-validation-total-months-of-cover.spec.js:96]: for example, I have no idea why it's 7.5.  but like const MONTH_WITH_DECIMAL would explain better?

Comment (ttbarnes) [reply e2e-tests/insurance/cypress/e2e/journeys/policy/multiple-contract-policy/validation/multiple-contract-policy-validation-total-months-of-cover.spec.js:96]: Is the test description not enough? "when total months of cover contains a decimal"

We have decimal tests like this everwhere with hard coded decimal values, e.g search for "contains a decimal"

Thing is, 7.5 is not month specific, it's just a number and so if we want this (and values with decimals) to be a constants, I suppose they should be generic, in which case how would a number constant be named?

**Review by Zainzzkk — APPROVED** [review]

**Pull request merged**